### DISCUSSION TITLE
Get entity by its ID

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6842] [General]
Hi all,

I need to get data of an entity by its ID to use them in my custom method:

```
    public class PecMailboxHelper
    {
        public static PecMailboxRow GetById(int pecId)
        {
            var pecMailbox = new  PecMailboxRow();

            // I need to get data from DB

            return pecMailbox;
        }
    }
```
I defined this class in a file under RequestHandlers folder, for now, but I want to move it in some BL folder

How can I do this?

Thank you,
Elena.

**Comment by luongnv6** [comment]
pls  use:
```C#
Connection.TryById<PecMailboxRow>(pecId);
```

**Reply by elenaaralla** [reply]
Hi @luongnv6,

it is the right way, thank you!
